android/services/mongodb-remote/src/main/java/com/mongodb/stitch/android/services/mongodb/remote/RemoteMongoClient.java

```diff
@@ -21,7 +21,7 @@
 import com.mongodb.stitch.android.services.mongodb.remote.internal.RemoteMongoClientImpl;
 import com.mongodb.stitch.core.StitchAppClientInfo;
 import com.mongodb.stitch.core.services.internal.CoreStitchServiceClient;
-import com.mongodb.stitch.core.services.mongodb.remote.internal.CoreRemoteMongoClientImpl;
+import com.mongodb.stitch.core.services.mongodb.remote.sync.internal.CoreRemoteClientFactory;
```

Comment:
what's this factory inside of sync?

Reply:
I might want to consider renaming it, but, it's the cleanest way to separate the logic for constructing a CoreRemoteMongoClient since now we need the necessary pieces for the DataSynchronizer. A lot of the constructors are package private.